web-js/src/components/useDraftTodos.js

```diff
@@ -0,0 +1,40 @@
+import React from "react"
+import * as Realm from "realm-web";
+
+export function useDraftTodos() {
+  const [drafts, setDrafts] = React.useState([]);
+
+  const createDraftTodo = () => {
+    const draftTodo = {
+      _id: new Realm.BSON.ObjectID(),
+      summary: "",
+      isComplete: false,
+    };
+    setDrafts((d) => [...d, draftTodo]);
+  };
+
+  const setDraftTodoSummary = (draft, summary) => {
```

Comment:
can you add some comments to this method (and actually probably all the methods in this file). Especially lines like: 
```
      return [
        ...oldDrafts.slice(0, idx),
        { ...oldDrafts[idx], summary },
        ...oldDrafts.slice(idx + 1),
      ];
```

and 

```
      const idx = oldDrafts.findIndex((d) => d._id === draft._id);
      return [...oldDrafts.slice(0, idx), ...oldDrafts.slice(idx + 1)];
```
These lines are hard to parse/scan quickly and **need** comments,